### ISSUE TITLE
Remove GSL as a hard dependency or add disclaimer

### ISSUE THREAD

**Issue by dev-zero** [electronic-structure/SIRIUS#298]
While SIRIUS is licensed under the FreeBSD-licensed (2-clause BSD), the GSL it needs to build is licensed under the GPL (not the **L**GPL!), which means that a binary SIRIUS installation must be distributed by the terms of the **GPL**.
Please note: I do not care either way, but if using the GSL was intentional, you might want to either change the license of SIRIUS to GPL itself or at least add a clarification that while the SIRIUS code itself is licensed as BSD, the resulting binary will be GPL due to this dependency.

**Comment by mtaillefumier** [comment]
Actually, we won't find any proper way to solve this because there is no agreement about the notion of derived work when linking is involved (it is not clearly stated in the license). Some say, it is derived work other no. In all honesty, I do not care that much about that based on the fact the code is licensed on a free license and in any case freely available for whom wanting it. Changing licence means agreement of all contributors (which as Joost knows can be a pain). So the option is to dual-licence it.

Basically, any project financed by public money only should be under at the very least BSD, or LGPL, GPL or equivalent. But dual licensing is also another option (and GPL does not forbid it).

**Comment by dev-zero** [comment]
@mtaillefumier wrt linking: https://www.gnu.org/licenses/gpl-faq.en.html#LinkingWithGPL ... so, from GNU side it is quiet clear and I wouldn't want to test that.

IANAL, but my understanding of this is:
The problem here is not your distribution of SIRIUS, but rather that every code incorporating SIRIUS has to comply to the GPL. This means that SIRIUS can not be used in any non-GPL compliant code and that every binary (and therefore also any source code making up that binary) must be made available under a GPL license. While this is not a problem for other GPL or more liberally licensed codes (if they are compatible in the first place, and since liberal licenses allow to tighten the conditions), it sure is for a non-FOSS code.

So, if you have chosen BSD to have just a FOSS license, then everything is fine and leaving a note stating that "due to linking to GSL which is GPL, the product will implicitly be GPL licensed" would be nice as a warning.
But if you have chosen a BSD license to make it possible to incorporate SIRIUS in a non-FOSS code, then you should remove the GSL, because that will not be possible.

**Comment by mtaillefumier** [comment]
This is not entirely true in practice and always comes back to the definition of derived work, which is not even clearly defined by law. Some people of fsf argue in the direction of the FAQs, others say that it does not explicitly violate the GPL provided that you publish any modification of the code licensed under GPL. Moreover, nothing forbid to use GPL software in a commercial software (which is not necessarily under the same license) provided that you are able to 

- provide the source code upon request
- publish any modification made to GPL parts that the code uses.

The LGPL provides solution for the linking issues, as for the rest using GPL stuff in BSD software does not necessarily breaks anything provided the two points mentioned above are respected. The inheritance although strongly suggested in the GPL license in general, is in practice not often taken as argument for laws suit unless you do not respect the others conditions explicitly indicated by the GPL. 

It is clearly not a problem in the context of cp2k since cp2k is GPL anyway.

**Comment by dev-zero** [comment]
If what you said were true there would be no need for the LGPL, but clearly some people make a distinction.

The problem for otherwise closed source codes is exactly that they would have to
* `provide the source code (to their software) upon request`
* **AND** under the terms of the GPL

which means allowing recipients of the source code to redistribute the softwares otherwise closed source code.

No matter how *we* are going to interpret it, any legal department will not allow using SIRIUS in a commercial code to avoid any possible issues.

As I wrote earlier: I don't care about it. I was simply surprised that the code was developed under a liberal license but requires a package under a strong copyleft license. Just wanted to ask whether that was intentional (in which case a warning would be nice, also for Linux distributions which have to track this stuff) or accidental.

**Comment by toxa81** [comment]
We had an internal discussion about licensing and we came to the conclusion that we continue to develop under BSD licence and potential non-FOSS code users/developers would have to take care of GSL themselves.
